Sources/SwiftSyntax/Syntax.swift

```diff
@@ -81,6 +81,76 @@ extension Syntax: CustomReflectable {
   }
 }
 
+extension Syntax: CustomDebugStringConvertible {
+  public var debugDescription: String {
+    return debugDescription()
+  }
+
+  public func debugDescription(includeChildren: Bool = true, includeTrivia: Bool = false,
+                               converter: SourceLocationConverter? = nil, mark: Syntax? = nil,
+                               indent: Int = 0) -> String {
+    var str = ""
+    debugWrite(to: &str, includeChildren: includeChildren, includeTrivia: includeTrivia, converter: converter, mark: mark, indent: indent)
+    return str
+  }
+
+  public func debugWrite<Target: TextOutputStream>(to target: inout Target,
+                                                   includeChildren: Bool = true, includeTrivia: Bool = false,
+                                                   converter: SourceLocationConverter? = nil, mark: Syntax? = nil,
+                                                   indent: Int = 0) {
+    var topLevelConverter = converter
+    if topLevelConverter == nil, let file = `as`(SourceFileSyntax.self) {
+      topLevelConverter = SourceLocationConverter(file: "", tree: file)
+    }
+
+    if self == mark {
+      target.write("*** ")
+    }
+
+    if isMissing {
+      target.write("- ")
+    }
```

Comment:
Not actually possible right now, but IMO walking missing (expected) children would make sense.